### ISSUE TITLE
Training loss abruptly changes

### ISSUE THREAD

**Issue by Mushtaqml** [facebookresearch/barlowtwins#18]
Hi!

Any idea what can be the reason for such jump in the training loss:

![image](https://user-images.githubusercontent.com/81066337/115819418-d7633580-a40f-11eb-8842-f2292d0b6ccf.png)

**Comment by liuhao-lh** [comment]
I also met this situation and want to know why.

**Comment by HareshKarnan** [comment]
i would check the "drop_last" parameter in your dataloader for the training set. If the batch_size is arbitrarily small, which happens when your # of datapoints is indivisible by batch_size, the last batch is arbitrarily small which messes the gradients.
